### ISSUE TITLE
Keyword only arguments type hints and highlights is not correct

### ISSUE THREAD

**Issue by paaksing** [microsoft/pylance-release#3515]
This issue have existed for a long period of time, but I guess my mental is not able to "not see" it anymore.

<h3> Environment data </h3>

-   Language Server version: 2022.10.30
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.8
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: off

<h3> Code Snippet </h3>

```python
from typing import TypedDict


class A(TypedDict):
    a: int
    b: str
    c: float


A(a=0, b="", c=0.0)


def x(x: int, *, y: float, z: str):
    return (x, y, z)

x(0, y=0.0, z="")

```

<h3> Repro Steps </h3>

1.  Place cursor on the keyword only arguments.

<h3> Expected behavior </h3>

The highlighted argument matches the one on the cursor.

<h3> Actual behavior </h3>

![image](https://user-images.githubusercontent.com/58193352/197291090-e7aa582a-076a-4880-94bc-31aa37d11f0a.png)
`y` ? it should be `z`

![image](https://user-images.githubusercontent.com/58193352/197291128-f64f2a47-a634-43f1-a5ad-78a763cc4d62.png)
seriously even typed dicts ?

<h3> Logs </h3>

No out of ordinary logs.

**Comment by bschnurr** [comment]
Thank you for the issue!

We were just talking about this.

**Comment by debonte** [comment]
Dupe of https://github.com/microsoft/pylance-release/issues/3507?

**Comment by bschnurr** [comment]
fixed in latest pre-release 2022.11.32
